internal/provider/inventory_resource.go

```diff
@@ -0,0 +1,319 @@
+	}
+
+	// Save latest inventory data into inventory resource model
+	diags = data.parseHTTPResponse(readResponseBody)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated state
+	diags = resp.State.Set(ctx, data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update updates the inventory resource and sets the updated Terraform state on success.
+func (r *InventoryResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data inventoryResourceModel
+	var diags diag.Diagnostics
+
+	// Read Terraform plan data into inventory resource model
+	diags = req.Plan.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Generate request body from inventory data
+	updateRequestBody, diags := data.generateRequestBody()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	requestData := bytes.NewReader(updateRequestBody)
+
+	// Update inventory in AAP
+	updateResponseBody, diags := r.client.Update(data.Url.ValueString(), requestData)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated inventory data into inventory resource model
+	diags = data.parseHTTPResponse(updateResponseBody)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated state
+	diags = resp.State.Set(ctx, data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Delete deletes the inventory resource.
+func (r *InventoryResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data inventoryResourceModel
+	var diags diag.Diagnostics
+
+	// Read current Terraform state data into inventory resource model
+	diags = req.State.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete inventory from AAP
+	_, diags = r.client.Delete(data.Url.ValueString())
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// InventoryResourceModel maps the inventory resource schema to a Go struct.
+type inventoryResourceModel struct {
+	Id           types.Int64          `tfsdk:"id"`
+	Organization types.Int64          `tfsdk:"organization"`
+	Url          types.String         `tfsdk:"url"`
+	Name         types.String         `tfsdk:"name"`
+	Description  types.String         `tfsdk:"description"`
+	Variables    jsontypes.Normalized `tfsdk:"variables"`
+}
+
+// generateRequestBody creates a JSON encoded request body from the inventory resource data.
+func (r *inventoryResourceModel) generateRequestBody() ([]byte, diag.Diagnostics) {
+	// Convert inventory resource data to API data model
+	var organizationId int64
+
+	// Use default organization if not provided
+	if r.Organization.ValueInt64() == 0 {
```

Comment:
not really sure if TF set the default value to `0` while not explicitly provided by the user.
I would rather suggest use both [IsNull()](https://github.com/hashicorp/terraform-plugin-framework/blob/79accaca46cc3989a3defffdc14e8ad3de8f111d/types/basetypes/int64_value.go#L132) and [IsUnknown()](https://github.com/hashicorp/terraform-plugin-framework/blob/79accaca46cc3989a3defffdc14e8ad3de8f111d/types/basetypes/int64_value.go#L137)

Reply:
https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework@v1.5.0/types/basetypes#Int64Value.ValueInt64

> If Int64 is null or unknown, returns 0.